sql/src/test/java/io/crate/protocols/postgres/PostgresNettyPublishPortTest.java

```diff
@@ -63,6 +73,61 @@ public void testNonUniqueBoundAddress() throws Exception {
             getByName("127.0.0.3"));
     }
 
+    @Test
+    public void testBindAndPublishAddressOverride() {
+        // First check if binding to a local works
+        NetworkService networkService = new NetworkService(Collections.emptyList());
+        PostgresNetty psql = new PostgresNetty(Settings.EMPTY, Mockito.mock(SQLOperations.class),
+            networkService, new AlwaysOKNullAuthentication(), Mockito.mock(SslContextProvider.class));
+        try {
+            psql.doStart();
+        } finally {
+            psql.doStop();
+        }
+
+        // Check override for network.host
+        Settings settingsWithCustomHost = Settings.builder().put("network.host", "cantbindtothis").build();
+        psql = new PostgresNetty(settingsWithCustomHost, Mockito.mock(SQLOperations.class), networkService,
```

Comment:
why are we putting different test cases into the same method if the setup is repeated and quite cheap anyway?

(also in the other tests - would easily be possible to create one test method per test case instead of bundling everything into a single method. It's easier to debug later on in case something breaks and also opens up the possibility to have more meaningful test names)

Reply:
The test resembles a logical test entity. Breaking it up is possible but scatters the test logic. If you think it is easier, I'll split it up.